### PR TITLE
Initial types to return from inspection of string

### PR DESCRIPTION
**Pull request by CJGutz**
closes: #8 

## Proposed changes
Added two files for types for the inspector

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added
- [ ] Documentation have been written

### CODE REVIEW

File: src/inspector/types/error.ts
Comment (thomsen85) [inline src/inspector/types/error.ts:1]: errors.ts instead?

File: src/inspector/types/error.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+export type ReasonedError = {
+  name: string;
+  message: string;
+  reason: ErrorReason;
+};
+
+export type Result<T, E = ReasonedError> = { kind: 'ok'; value: T } | { kind: 'error'; error: E };
+// | { kind: "partial" ; value: Partial<T>; error: E };
```
Comment (thomsen85) [inline src/inspector/types/error.ts:14]: Remove?

**Review by thomsen85 — APPROVED** [review]
Looks good, just some nit-picks

File: src/inspector/types/error.ts (hunk 1 above)
Comment (CJGutz) [reply src/inspector/types/error.ts]: Yes agreed

File: src/inspector/types/error.ts
Comment (CJGutz) [reply src/inspector/types/error.ts:1]: Yes agreed

**Pull request merged**